### PR TITLE
Pass in user_id to workshop enroll form

### PR DESCRIPTION
**Pull request by TurnerRiley**
While working on [this PR](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-508), we were having trouble efficiently obtaining the user for a given enrollment in the workshop enrollment flow. Since viewing the enroll page for a workshop requires that the user is signed in and has an application, we can safely pass in the `user_id` and make it required.

### Demo that going through the enroll form flow still works:
https://github.com/code-dot-org/code-dot-org/assets/56283563/3a97d52d-eb0b-4348-a42c-14836217e291

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-1091)
PR for use of alternate email that builds on this: [here](https://github.com/code-dot-org/code-dot-org/pull/53437)

## Testing story
Updating unit tests and local testing.

## Follow-up work
Simplify the [aformentioned PR](https://github.com/code-dot-org/code-dot-org/pull/53437) to use `user_id` to get the user.

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_enrollment/workshop_enroll.jsx [hunk 1]
```diff
@@ -23,6 +23,7 @@ const SUBMISSION_STATUSES = {
 
 export default class WorkshopEnroll extends React.Component {
   static propTypes = {
+    user_id: PropTypes.number.isRequired,
```
Comment (bethanyaconnor) [inline apps/src/code-studio/pd/workshop_enrollment/workshop_enroll.jsx:26]: where is this prop coming from? said another way, does a different component need to be updated to pass this through?

Comment (TurnerRiley) [reply apps/src/code-studio/pd/workshop_enrollment/workshop_enroll.jsx:26]: This is passed in from the `workshop_enrollment_controller` here: https://github.com/code-dot-org/code-dot-org/pull/54826/files#diff-658aa36aa0f026bca4b58f798c1e3bdf2fa93625441c01de0f588dbcdc6a678aR74

Comment (bethanyaconnor) [reply apps/src/code-studio/pd/workshop_enrollment/workshop_enroll.jsx:26]: oof that's confusing 😅 but definitely out of the scope of this PR to fix so LGTM!

**Review by bethanyaconnor — APPROVED** [review]

File: apps/src/code-studio/pd/workshop_enrollment/workshop_enroll.jsx (hunk 1 above)
Comment (TurnerRiley) [reply apps/src/code-studio/pd/workshop_enrollment/workshop_enroll.jsx:26]: Yes totally agree! Sounds good thank you!

**Pull request merged**